cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTable.java

```diff
@@ -125,7 +134,8 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
                                                  callerRunsPolicy);
 
       this.rowKeyDistributor = new RowKeyDistributorByHashPrefix(
-        new RowKeyDistributorByHashPrefix.OneByteSimpleHash(Constants.Metrics.METRICS_HBASE_SPLITS));
+        new RowKeyDistributorByHashPrefix.
+          OneByteSimpleHash(spec.getIntProperty(Constants.Metrics.METRICS_HBASE_TABLE_SPLITS, 16)));
```

Comment:
because the dataset is always created with this property - do you need the default here? If you pass in a default, it should be the value from cConf. I think that can only happen when people upgrade from 4.3RC to 4.3GA - their metrics tables would have been created without the property. Oh is that why you use 16 as the default?